x/dispensation/keeper/distributionRecords.go

```diff
@@ -31,95 +31,111 @@ func (k Keeper) GetDistributionRecord(ctx sdk.Context, airdropName string, recip
 	return &dr, nil
 }
 
-func (k Keeper) ExistsDistributionRecord(ctx sdk.Context, airdropName string, recipientAddress string) bool {
-	key := types.GetDistributionRecordKey(airdropName, recipientAddress)
+func (k Keeper) ExistsDistributionRecord(ctx sdk.Context, airdropName string, recipientAddress string, status types.DistributionStatus) bool {
+	key := types.GetDistributionRecordKey(status, airdropName, recipientAddress)
 	if k.Exists(ctx, key) {
 		return true
 	}
 	return false
 }
 
-func (k Keeper) GetDistributionRecordsIterator(ctx sdk.Context) sdk.Iterator {
+func (k Keeper) GetDistributionRecordsIterator(ctx sdk.Context, status types.DistributionStatus) sdk.Iterator {
 	store := ctx.KVStore(k.storeKey)
-	return sdk.KVStorePrefixIterator(store, types.DistributionRecordPrefix)
+	switch status {
+	case types.DistributionStatus_DISTRIBUTION_STATUS_PENDING:
+		return sdk.KVStorePrefixIterator(store, types.DistributionRecordPrefixPending)
+	case types.DistributionStatus_DISTRIBUTION_STATUS_COMPLETED:
+		return sdk.KVStorePrefixIterator(store, types.DistributionRecordPrefixCompleted)
+	default:
+		return sdk.KVStorePrefixIterator(store, types.DistributionRecordPrefixCompleted)
```

Comment:
Is this a useful default or is it not better to return an iterator for unspecified status?